### DISCUSSION TITLE
assign duplicating workspace instead of assigning the app to the workspace

### DISCUSSION THREAD

**Discussion by barnaba** [regolith-linux/regolith-desktop#831] [Q&A]
Hi,

I'm trying to assign discord to my second workspace with:
`assign [class="discord"] "$ws2"`

This works, as discord starts in workspace 2. The problem is, it's a different workspace 2 than the workspace 2 I already had - resulting in 2 workspaces 2: 
![image](https://user-images.githubusercontent.com/179017/224553673-84e1ee57-3472-4fc9-8573-bf68732f0672.png)

As long as this new workspace 2 exists shift+mod+2 moves other apps to this workspace (and doesn't move apps from the new workspace 2 to the old workspace 2). The only way to move the apps to old workspace 2 is to move them all to some other workspace (e.g. 1) untill the empty new workspace 2 stops existing and then I mod+shift+2 moves them to the old workspace 2 again.

Alternatively I can move all my apps from the old workspace 2 to the new one using shift+mod+2 and from that point everything works in that session (including a case in which display 2 disappears and I recreate it by mod+shift+2 and run discord later)

**Comment by kgilmer** [comment]
I'm guessing there is some whitespace in one of the workspace labels that causes i3 to consider it to be distinct.  You may see if listing the i3 workspaces with `i3-msg` when you see this duplicate leads to any clarity..

**Comment by barnaba** [comment]
Thanks.
they seem to use different string quotes.

```json
[
  {
    "id": 94463326721776,
    "num": 2,
    "name": "2:<span font_desc='BitstreamVeraSansMono Nerd Font 12'> 2 </span>",
...
  },
  {
    "id": 94463326487920,
    "num": 2,
    "name": "2:<span font_desc=\"BitstreamVeraSansMono Nerd Font 12\"> 2 </span>",
...
  },
```
  
I suppose the way i3 compares the names and allows for multiple displays with the same num is an i3 problem/feature that Regolith won't solve. IDK if it's in Regolith's scope to make ` assign [class="discord"] "$ws2"` work or if it's even possible. 

I have solved my initial problem by using workspace number instead of name:
``` assign [class="discord"] "$ws2"```
to
```assign [class="discord"] workspace number $ws2```. 

This makes me happy and I don't know if or how the underlying name difference can be solved in Regolith (and I wasn't able to trace where it might even be coming from). If it's something that can be improved I'll be happy to assist - if not **feel free to close** the issue as solved.
